src/main/app/case/to-api-format.test.ts

```diff
@@ -333,4 +342,57 @@ describe('to-api-format', () => {
   ])('set unreachable answers to null if condition met', ({ expected, ...formData }) => {
     expect(toApiFormat(formData as Partial<Case>)).toMatchObject(expected);
   });
+
+  test.each([
+    {
+      applicant1ApplyForFinancialOrder: YesOrNo.YES,
+      applicant1WhoIsFinancialOrderFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      applicant2ApplyForFinancialOrder: YesOrNo.YES,
+      applicant2WhoIsFinancialOrderFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      expected: {
+        applicant1FinancialOrder: YesOrNo.YES,
+        applicant1FinancialOrdersFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+        applicant2FinancialOrder: YesOrNo.YES,
+        applicant2FinancialOrdersFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      },
+    },
+    {
+      applicant1ApplyForFinancialOrder: YesOrNo.NO,
+      applicant1WhoIsFinancialOrderFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      applicant2ApplyForFinancialOrder: YesOrNo.NO,
+      applicant2WhoIsFinancialOrderFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      expected: {
+        applicant1FinancialOrder: YesOrNo.NO,
+        applicant1FinancialOrdersFor: [],
+        applicant2FinancialOrder: YesOrNo.NO,
+        applicant2FinancialOrdersFor: [],
+      },
+    },
+    {
+      applicant1ApplyForFinancialOrder: YesOrNo.NO,
+      applicant1WhoIsFinancialOrderFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      applicant2ApplyForFinancialOrder: YesOrNo.YES,
+      applicant2WhoIsFinancialOrderFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      expected: {
+        applicant1FinancialOrder: YesOrNo.NO,
+        applicant1FinancialOrdersFor: [],
+        applicant2FinancialOrder: YesOrNo.YES,
+        applicant2FinancialOrdersFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      },
+    },
+    {
+      applicant1ApplyForFinancialOrder: YesOrNo.YES,
+      applicant1WhoIsFinancialOrderFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      applicant2ApplyForFinancialOrder: YesOrNo.NO,
+      applicant2WhoIsFinancialOrderFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+      expected: {
+        applicant1FinancialOrder: YesOrNo.YES,
+        applicant1FinancialOrdersFor: [FinancialOrderFor.APPLICANT, FinancialOrderFor.CHILDREN],
+        applicant2FinancialOrder: YesOrNo.NO,
+        applicant2FinancialOrdersFor: [],
+      },
+    },
```

Comment:
applicant1ApplyForFinancialOrder and applicant2ApplyForFinancialOrder are unrelated so how comes your testing Yes/No and No/Yes situations? I would have thought Yes Yes and No No would be enough

Reply:
in the to-api-format, if they select no, it sets the array to empty, hence im doing different situations.

Reply:
but you're already testing that no situation in the test above?

Reply:
just thought ill add extra test to indicate one doesnt effect the other.
so saying no to one doesnt set the other applicants array.
i can remove it if you prefer, i was just covering different situations.

Reply:
yeah seems a bit unnecessary to me